### PR TITLE
SMLWIN21-42 Selenium framework with basic test

### PR DESCRIPTION
**Pull request by sharonO13**
Basic Automation Framework with one test for navigation to application and title test
Consists of 

- Webdriverhelper class to manage the webdriver as well as additional screen shot functionality
- BasePage for managing WebElements
- Page Object Model approach with a class defining page objects
- Testcase for test execution with before and after to initiate and teardown the webdriver

### CODE REVIEW

File: test/ui/testautomation/src/main/java/smallwindow/utils/WebDriverHelper.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.TakesScreenshot;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.remote.CapabilityType;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.io.FileHandler;
+import org.openqa.selenium.*;
+
+public final class WebDriverHelper{
```
Comment (L00163105) [inline test/ui/testautomation/src/main/java/smallwindow/utils/WebDriverHelper.java:29]: We are missing Javadoc

File: test/ui/testautomation/src/main/java/smallwindow/utils/WebDriverHelper.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        } 
+        
+        return null;
+    }
+
+    public static File getScreenshot(WebDriver driver){
+        File screen_shot = null;
+
+        try{
+            if(driver == null){
```
Comment (L00163105) [inline test/ui/testautomation/src/main/java/smallwindow/utils/WebDriverHelper.java:159]: Dont think this `if` is needed, as its potentially returning null (`File screen_shot = null;`). The catch will handle it below and return 'screen_shot = null' anyway

File: test/ui/testautomation/src/main/java/smallwindow/pages/HomePage.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import org.openqa.selenium.By;
+
+public class HomePage extends BasePage{
+
+    public HomePage(WebDriver driver){
+        super(driver);
+    }
+
+    //list of page objects and methods
+    public WebElement NavBarElement(){
```
Comment (L00163105) [inline test/ui/testautomation/src/main/java/smallwindow/pages/HomePage.java:14]: `NavBarElement` > `navBarElement`. Sonar should pick up this stuff if we have it configured to scan the testautomation directory?

**Comment by sharonO13** [comment]
Updates applied

**Review by L00162772 — APPROVED** [review]
Looks good Sharon, nice job

File: test/ui/testautomation/src/main/java/smallwindow/utils/WebDriverHelper.java (hunk 1 above)
Comment (sharonO13) [reply test/ui/testautomation/src/main/java/smallwindow/utils/WebDriverHelper.java:36]: Javadoc added

**Pull request merged**